subsys/shell/shell.c

```diff
@@ -599,10 +599,11 @@ static int execute(const struct shell *shell)
 
 	memset(&shell->ctx->active_cmd, 0, sizeof(shell->ctx->active_cmd));
 
-	shell_cmd_trim(shell);
-
-	history_put(shell, shell->ctx->cmd_buff,
-		    shell->ctx->cmd_buff_len);
+	if (IS_ENABLED(CONFIG_SHELL_HISTORY)) {
+		shell_cmd_trim(shell);
+		history_put(shell, shell->ctx->cmd_buff,
```

Comment:
`history_put` is a local function for shell.c file. I can of course rename it but there are many other local functions in this file that do not have `shell_` prefix. Shall I update all of them?

Reply:
My point was that `history_put` differs from `shell_history_put` only by a `IS_ENABLED(CONFIG_SHELL_HISTORY)` check inside. In that case there would be no point of keeping `history_put()` anymore, because `shell_history_put()` could be used with the same effect. However looking closer at it, it seems that those local functions tend to improve readability. So maybe lets stick with `history_put()` for now.